src/framework/mpas_io_streams.F

```diff
@@ -4362,4 +4362,56 @@ subroutine put_get_field_atts(handle, ioDirection, fieldname, attList)
 
    end subroutine put_get_field_atts
 
+   function mpas_field_list_get_largest_field_size(fieldList, dmInfo) result(maxSize)
+      implicit none
+      type (field_list_type), pointer :: fieldList
+      type (dm_info) :: dmInfo
+
+      integer(kind=I8KIND) :: maxSize
+      integer(kind=I8KIND) :: fieldSize
+      character (len=strKIND) :: fieldNameOut
+      type (field_list_type), pointer :: fieldCursor
+
+      maxSize = 0
+
+      if (.not. associated(fieldList)) return
+
+      fieldCursor => fieldList
+
+      do while (associated(fieldCursor))
+         select case (fieldCursor % field_type)
+            case (FIELD_0D_INT)
+               call mpas_dmpar_sum_int8(dmInfo, sizeof(fieldCursor % int0dField % scalar), fieldSize)
```

Comment:
With the GNU compilers and `-std=f2008`, this code will fail to compile:
```
mpas_io_streams.F:4384:48:

 4384 |                call mpas_dmpar_sum_int8(dmInfo, sizeof(fieldCursor % int0dField % scalar), fieldSize)
      |                                                1
Error: Function ‘sizeof’ at (1) has no IMPLICIT type
```
The `sizeof` function is a GNU extension: https://gcc.gnu.org/onlinedocs/gfortran/SIZEOF.html